org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/testexecution/loglines/Log4JDefaultFilterTest.xtend

```diff
@@ -0,0 +1,91 @@
+package org.testeditor.web.backend.testexecution.loglines
+
+import java.util.Collection
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+
+import static org.assertj.core.api.Assertions.assertThat
+import static org.testeditor.web.backend.testexecution.loglines.LogLevel.*
+
+@RunWith(Parameterized)
+@FinalFieldsConstructor
+class Log4JDefaultFilterTest {
+
+	val String logLine
+	val LogLevel logLevel
+	val boolean expectedResult
+
+	static val logLines = #{
+		CRITICAL -> '    23:42:00 FATAL [Test worker]  [TE-Test: TestName] ReportingClass Message part.',
+		ERROR -> '    23:42:00 ERROR [Test worker]  [TE-Test: TestName] ReportingClass Message part.',
+		WARNING -> '    23:42:00 WARN  [Test worker]  [TE-Test: TestName] ReportingClass Message part.',
+		INFO -> '    23:42:00 INFO  [Test worker]  [TE-Test: TestName] ReportingClass Message part.',
+		DEBUG -> '    23:42:00 DEBUG [Test worker]  [TE-Test: TestName] ReportingClass Message part.',
+		TRACE -> '    23:42:00 TRACE [Test worker]  [TE-Test: TestName] ReportingClass Message part.'
+	}
+	static val unformattedLogLine = 'FATAL ERROR WARN INFO DEBUG TRACE unformatted log line'
+
+	@Parameters(name='log line: "{0}"; log level: {1}; accept? {2}')
+	def static Collection<Object[]> data() {
+		return #[
+			#[logLines.get(CRITICAL), CRITICAL, true],
```

Comment:
I would expect the result of isVisibleOn to return exactly the opposite actually. If I ask for CRITICAL I would expect only critical messages, if I ask for Errors, I'd expect CRITICAL and ERROR, asking for INFO, I'd expect CRITICAL, ERROR, WARNING and INFO (but no DEBUG and no TRACE).

Reply:
I think that is what the method does… isn't it?
Notice that the first argument is the actual log entry, and the second is the desired log level. This allows to write the method "extension style" and make it more readable (or so I thought :)
```
'23:42:00 WARN [Test worker] …'.isVisibleOn(INFO) // true, WARN is more severe than INFO
'23:42:00 WARN [Test worker] …'.isVisibleOn(WARN) // true
'23:42:00 WARN [Test worker] …'.isVisibleOn(ERROR) // false, WARN is less severe than ERROR
```

Reply:
Ahh, I finally got it